tests/workflows/ph/test_base.py

```diff
@@ -257,3 +257,23 @@ def test_merge_outputs(
     outputs = ph_base.get_outputs(node_2)
 
     data_regression.check(outputs['output_parameters'].get_dict())
+
+
+def test_validate_inputs_excluded_qpoints_distance(generate_inputs_ph):
+    """Test that q-points validation passes in case the ports are excluded in a parent work chain."""
+    from aiida.engine.utils import instantiate_process
+    from aiida.manage.manager import get_manager
+
+    class WrapPhBaseWorkChain(WorkChain):
+        """Minimal work chain that wraps a ``PhBaseWorkChain`` for testing excluding q-points inputs."""
+
+        @classmethod
+        def define(cls, spec):
+            super().define(spec)
+            spec.expose_inputs(PhBaseWorkChain, exclude=('qpoints', 'qpoints_distance'))
+
+    inputs = {'ph': generate_inputs_ph()}
+    inputs['ph'].pop('qpoints', None)
+    inputs['ph'].pop('qpoints_distance', None)
```

Comment:
I don't think this is really needed right? Since it is an input of the PhBaseWorkChain

Reply:
I mean the last line of the one highlighted.

Reply:
Oh fair, it's not in there anyways 👍

Reply:
> I don't think this is really needed right? Since it is an input of the PhBaseWorkChain

I don't follow. That is the whole point of this test. It is checking that, even though `qpoints` and `qpoints_distance` _are_ inputs to `PhBaseWorkChain` and so they are validated in the inputs namespace validator, when a wrapping workchain explicitly excludes them, the validator does not raise a warning. So the test needs to make sure that inputs are passed that do _not_ contain either `qpoints` or `qpoints_distance`. Right?

Reply:
Correct, but `generate_inputs_ph()` would generate inputs of `PhCalculation`, which doesn't have `qpoints_distance`, so it's one line more which is not needed. The two qpoints input for `PhBaseWorkChain` are moved into the "top level" namespace, and not nested in the `ph` namespace.

Reply:
Aaaah, I see, do you mean that since I used `generate_inputs_ph` which generates inputs for `PhCalculation` and not the workchain, there won't be a `qpoints_distance` to begin with. That makes sense. But then I think the real fix should be to replace `generate_inputs_ph` with a fixture that generates the inputs for `PhBaseWorkChain` and then pop the relevant inputs. This would be more future proof to changes in the input specs of the processes and the input generating fixtures.

Reply:
Yeah but I used the `generate_inputs_ph` fixture which wouldn't have the `qpoints_distance` (there is no `generate_inputs_ph_base` afaik).

Reply:
Fair enough, I do agree.

Reply:
I think one can use `generate_ph_workchain_node(return_inputs=True)` or similar (don't remember exactly the name of the fixutre)

Reply:
Ok, feel free to change as I suggested, but not necessary. Also fine to merge as is for me